webclient/index.html

```diff
@@ -63,8 +63,18 @@
       })();
 
       chatApi.onMessageReceived(function(messageText) {
+        // todo: learn why these types of operations are expensive and how to
+        // mitigate those expenses;
         // Add a new message to the list of messages
         $('#messages').append($('<li class="list-group-item">').text(messageText));
+        
+        var $viewport = $('.viewport')[0];
+        var scrollTop = $viewport.scrollTop;
+        var scrollHeight = $viewport.scrollHeight;
+        var viewportHeight = $viewport.clientHeight;
+        if(scrollHeight -  scrollTop - viewportHeight <= 200) {
+          $viewport.scrollTop = scrollHeight;
+        }
```

Comment:
Nice! This is the sort of automatic scrolling behavior that I was hoping for. The behavior here is that when a new message is received, it will scroll the chat view all the way to the bottom, but only if the user is already sufficiently scrolled towards the bottom. Anyone who isn't scrolled to the bottom is reading towards the top, and  it's good that they're unaffected because changing their scroll level would not be a good reading experience.